### PR TITLE
PLT-6450 Migrate installed_outgoing_webhook.jsx to be pure and use Redux

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
Migrate installed_outgoing_webhook.jsx to be pure and use Redux

#### Ticket Link
https://github.com/mattermost/platform/issues/6450

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: webapp/components/integrations/components/installed_outgoing_webhook.jsx [hunk 1]
```diff
@@ -69,28 +111,19 @@ export default class InstalledOutgoingWebhook extends React.Component {
...
-            displayName = channel.display_name;
-        } else {
-            displayName = (
-                <FormattedMessage
-                    id='installed_outgoing_webhooks.unknown_channel'
-                    defaultMessage='A Private Webhook'
-                />
-            );
-        }
+        const displayName = InstalledOutgoingWebhook.makeDisplayName({
```
Comment (hmhealey) [inline webapp/components/integrations/components/installed_outgoing_webhook.jsx]: Do you mind changing this to just take those two arguments without passing them in as an object? We usually just pass them in directly unless there's a lot of arguments or default values being used.

File: webapp/components/integrations/components/installed_outgoing_webhooks.jsx [hunk 2]
```diff
@@ -115,6 +115,7 @@ export default class InstalledOutgoingWebhooks extends React.Component {
                     creator={this.state.users[outgoingWebhook.creator_id] || {}}
                     canChange={canChange}
                     team={this.props.team}
+                    channel={channel}
```
Comment (hmhealey) [inline webapp/components/integrations/components/installed_outgoing_webhooks.jsx]: Similar thing here to @jwilander's on your other PR. `installed_incoming_webhook.jsx` should be moved into `installed_incoming_webhook/installed_incoming_webhook.jsx` and then you should create a container in  `installed_incoming_webhook/index.jsx` that connects to the redux store to get the channel.

File: webapp/components/integrations/components/installed_outgoing_webhook.jsx (hunk 1 above)
Comment (sudheerDev) [inline webapp/components/integrations/components/installed_outgoing_webhook.jsx]: @hmhealey ya sure, I agree i don't think passing as an object makes much of sense here.

File: webapp/components/integrations/components/installed_outgoing_webhooks.jsx (hunk 2 above)
Comment (sudheerDev) [inline webapp/components/integrations/components/installed_outgoing_webhooks.jsx]: @jwilander was mentioning in the second comment about keeping` installed_incoming_webhook` the way as it exists in PR. Change `installed_incoming_webhooks` to be a connected component with the channel as a selector and pass it as props to `installed_incoming_webhook`.

Comment (sudheerDev) [inline webapp/components/integrations/components/installed_outgoing_webhooks.jsx]: If this makes sense then https://github.com/mattermost/platform/issues/6454 commit should resolve this PR.

File: webapp/components/integrations/components/installed_outgoing_webhook.jsx (hunk 1 above)
Comment (sudheerDev) [inline webapp/components/integrations/components/installed_outgoing_webhook.jsx]: @hmhealey done

**Comment by crspeller** [comment]
This PR needs to be re-submitted to the new separated webapp and server repositories.

**Pull request closed without merging**